src/js/language_label.js

```diff
@@ -1,15 +1,179 @@
 "use strict";
 
-var langField = document.createElement("div");
-langField.id = "language-field";
+import Tokenfield from "tokenfield";
+import { updateLanguageLabel } from "../americana";
+import * as Label from "../constants/label.js";
+
+var langField = labelControlElement("span", "language-field");
+
+var langChanger = labelControlElement("button", "language-switcher");
+langChanger.textContent = "Change";
+
+var langPicker = labelControlElement("input", "language-picker");
+hide(langPicker);
+
+var langHeader = labelControlElement("span", "lang-header");
+var langHints = labelControlElement("span", "lang-hints");
+langHints.textContent = "Begin typing to add languages";
+
+var langCancel = labelControlElement("button", "language-cancel");
+langCancel.textContent = "X";
+Object.assign(langCancel.style, {
+  "margin-top": "0.3em",
+});
+
+langHeader.appendChild(langCancel);
+langHeader.appendChild(langHints);
+
+hide(langHeader);
+
+function hide(element) {
+  Object.assign(element.style, {
+    display: "none",
+  });
+}
+function show(element) {
+  element.style.removeProperty("display");
+}
+
+let languageNames = new Intl.DisplayNames(Label.getLocales(), {
+  type: "language",
+});
+let langCodes = [
+  "am",
+  "ar",
+  "az",
+  "be",
+  "bg",
+  "br",
+  "bs",
+  "ca",
+  "co",
+  "cs",
+  "cy",
+  "da",
+  "de",
+  "el",
+  "en",
+  "eo",
+  "es",
+  "et",
+  "eu",
+  "fi",
+  "fr",
+  "fy",
+  "ga",
+  "gd",
+  "he",
+  "hi",
+  "hr",
+  "hu",
+  "hy",
+  "id",
+  "is",
+  "it",
+  "ja",
+  // "ja_kana",
+  "ja-Latn",
+  "ja-Hira",
+  "ka",
+  "kk",
+  "kn",
+  "ko",
+  "ko-Latn",
+  "ku",
+  "la",
+  "lb",
+  "lt",
+  "lv",
+  "mk",
+  "mt",
+  "ml",
+  "nl",
+  "no",
+  "oc",
+  "pl",
+  "pt",
+  "rm",
+  "ro",
+  "ru",
+  "sk",
+  "sl",
+  "sq",
+  "sr",
+  "sr-Latn",
+  "sv",
+  "ta",
+  "te",
+  "th",
+  "tr",
+  "uk",
+  "zh",
+].map((id) => {
+  return { id, name: languageNames.of(id) };
+});
+
+function labelControlElement(tag, id) {
+  var element = document.createElement(tag);
+  element.id = id;
+  Object.assign(element.style, {
+    margin: "0 2.5px",
+    color: "#444",
+  });
+  return element;
+}
+
+var tf = null;
+
+langChanger.onclick = function () {
+  hide(langChanger);
+  show(langHeader);
+
+  if (tf == null) {
+    tf = new Tokenfield({
+      el: document.querySelector("#language-picker"), // Attach Tokenfield to the input element with class "text-input"
+      items: langCodes,
+      newItems: false,
+    });
+    document.querySelectorAll(".tokenfield").forEach((e) => {
+      Object.assign(e.style, {
+        height: "5em",
+        width: "20em",
+        "margin-bottom": "4em",
+        "margin-top": "0.3em",
```

Comment:
I only see a couple of the suggestions at a time, because the suggestion menu only appears below the textbox that’s already pretty far down the page. If Tokenfield doesn’t support automatically flipping the menu above the textbox, we should find a different place for the setting as tail work.

Reply:
Does not appear to have that option unfortunately:
https://github.com/KaneCohen/tokenfield